src/buildercore/cfngen.py

```diff
@@ -406,7 +406,10 @@ def merge_delta(stackname, delta_plus, delta_minus):
 def apply_delta(template, delta_plus, delta_minus):
     for component in delta_plus:
         ensure(component in ["Resources", "Outputs"], "Template component %s not recognized", component)
-        template[component].update(delta_plus[component])
```

Comment:
really minor thing, but this:
```
data = template.get(component, {})
data.update(delta_plus[component])
template[component] = data
```
avoids any forking in logic and coverage overhead. I don't think there is any complexity checking in builder yet.